keadm/cmd/keadm/app/cmd/helm/cloudcore.go

```diff
@@ -0,0 +1,250 @@
+	if err != nil {
+		return fmt.Errorf("failed to get cloudcore history config, err: %v", err)
+	}
+
+	componentName := CloudCoreHelmComponent
+	subDir := fmt.Sprintf("%s/%s", ChartsSubDir, CloudCoreHelmComponent)
+
+	helper, err := NewHelper(opts.KubeConfig, constants.SystemNamespace)
+	if err != nil {
+		return err
+	}
+	// Determine whether the cloudcore release has been installed
+	if rel, err := helper.GetRelease(componentName); err != nil {
+		return err
+	} else if rel == nil {
+		return fmt.Errorf("the cloudcore release not found, and you can init the cloudcore using the `keadm init`")
+	}
+
+	// Load helm values or profile values, and combine the sets flag
+	vals, err := LoadHelmValues("", opts.Profile)
+	if err != nil {
+		return err
+	}
+	installed, err := helper.GetValues(componentName)
+	if err != nil {
+		return err
+	}
+	setImageTagOfComponents(c.ToolVersion.String(), opts.CloudInitUpdateBase)
+	valsmap, err := CombineValues(vals, installed, opts.GetValidSets())
+
+	// Build a new renderer instance
+	renderer := NewGenericRenderer(kecharts.BuiltinOrDir(""),
+		subDir, componentName, constants.SystemNamespace, valsmap, false)
+	// Load the charts to this renderer
+	if err := renderer.LoadChart(); err != nil {
+		return fmt.Errorf("cannot load the given charts %s, error: %s", renderer.componentName, err.Error())
+	}
+
+	// Upgrade the cloudcore chart
+	rel, err := helper.Upgrade(renderer, componentName, opts.DryRun, opts.Force)
+	if err != nil {
+		return err
+	}
+	// Print upgradation successful message
+	var lastDeployed string
+	if !rel.Info.LastDeployed.IsZero() {
+		lastDeployed = rel.Info.LastDeployed.Format(time.ANSIC)
+	}
+	fmt.Printf(MessageFormatInstallationSuccess,
+		strings.ToTitle(renderer.componentName),
+		rel.Name,
+		lastDeployed,
+		rel.Namespace,
+		rel.Info.Status.String(),
+		rel.Version)
+	fmt.Printf(MessageFormatUpgradationPrintConfig, cloudcoreConfig)
+	return nil
+}
+
+// Uninstall uses helm client to uninstall cloudcore release
+func (c *CloudcoreHelmTool) Uninstall(opts *types.ResetOptions) error {
+	// clean kubeedge namespace
+	if err := c.Common.CleanNameSpace(constants.SystemNamespace, opts.Kubeconfig); err != nil {
+		return fmt.Errorf("failed to clean kubeedge namespace, err: %v", err)
+	}
+	return nil
+}
+
+func (c *CloudcoreHelmTool) verifyCloudcoreOrocessRunning() error {
+	cloudcoreRunning, err := c.Common.OSTypeInstaller.IsKubeEdgeProcessRunning(util.KubeCloudBinaryName)
+	if err != nil {
+		return fmt.Errorf("failed to verify the cloudcore binnary already running, err: %v", err)
+	}
+	if cloudcoreRunning {
+		return fmt.Errorf("the cloudcore is already running on this node, please run reset to clean up first")
+	}
+	return nil
+}
+
+// setImageTagOfComponents when the component image tag is not set, it is set to a default value
+func setImageTagOfComponents(version string, opts types.CloudInitUpdateBase) {
+	if !opts.HasSets(setsKeyCloudcoreImageTag) {
```

Comment:
This block can use array `[]string{setsKeyCloudcoreImageTag,setsKeyControllerManagerImageTag,...}` for for loop operation

Reply:
Done, thanks